torch/_inductor/compile_fx.py

```diff
@@ -355,6 +357,23 @@ def compile_fx(
 
     functorch.compile.config.use_functionalize = True
     functorch.compile.config.use_fake_tensor = True
+    
+    fake_mode = None
+
+    flat_inputs, _ = tree_flatten(example_inputs_)
+    for input in flat_inputs:
+        if isinstance(input, torch._subclasses.FakeTensor):
+            if fake_mode is None:
+                fake_mode = input.fake_mode
+            else:
+                assert fake_mode == input.fake_mode
+
+    # if fake_mode:
+        # NOTE: This *will* create guards - we are missng
+        # the shape env guard supression logic here.
+        # We need to add it once its pushed up.
+        # Don't land this without that logic. 
+        # model_ = deepcopy_to_fake_tensor(model_, fake_mode)
```

Comment:
TODO I guess this part is WIP

Reply:
Whole PR is WIP :) but yes this is for sure